projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js

```diff
@@ -2,33 +2,91 @@
  * WordPress dependencies
  */
 import { useSelect, useDispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
 
 /*
  * Internal dependencies
  */
 import getJetpackExtensionAvailability from '../../../../shared/get-jetpack-extension-availability';
-import { isUpgradable, isUpgradeNudgeEnabled } from '../../../../shared/plan-utils';
+import { isUpgradable } from '../../../../shared/plan-utils';
 
 const republicizeFeatureName = 'republicize';
 
 export default function usePublicizeConfig() {
-	// Actions.
 	const { togglePublicizeFeature } = useDispatch( 'jetpack/publicize' );
-	const { available } = getJetpackExtensionAvailability( republicizeFeatureName );
+	const { available: isRePublicizeFeatureAvailable } = getJetpackExtensionAvailability(
+		republicizeFeatureName
+	);
+	const isPostPublished = useSelect( select => select( editorStore ).isCurrentPostPublished(), [] );
 
-	// Data.
-	const isPublicizeEnabled = useSelect(
+	/*
+	 * isPublicizeEnabledMeta:
+	 * It's stored in the jetpack_publicize_feature_enabled post metadata,
+	 * and usually is handled from the UI (main toggle control),
+	 * dispathicng the togglePublicizeFeature() action (jetpack/publicize).
+	 */
+	const isPublicizeEnabledMeta = useSelect(
 		select => select( 'jetpack/publicize' ).getFeatureEnableState(),
 		[]
 	);
 
+	/*
+	 * isRePublicizeFeatureEnabled:
+	 * Feature flag, defined by the server-side.
+	 * it can be defined via the `jetpack_block_editor_republicize_feature` backend filter.
+	 */
+	const isRePublicizeFeatureEnabled = !! window?.Jetpack_Editor_Initial_State.jetpack
+		?.republicize_enabled;
+
+	/*
+	 * isRePublicizeUpgradableViaUpsell:
+	 * True when the republicize feature is upgradable according to the store product (republicize),
+	 * but also whether the upgrade nudge is enable
+	 * in the site context/platform (Simple, Atomic, Jetpack, etc...).
+	 */
+	const isRePublicizeUpgradableViaUpsell = isUpgradable( republicizeFeatureName );
+
+	/*
+	 * isPublicizeEnabled:
+	 * Althought the feature is enabled by the post meta,
+	 * it also depends on whether the product feature.
+	 * Also, it's tied to the post status (draft, published, etc.).
+	 */
+	const isPublicizeEnabled =
+		( isPublicizeEnabledMeta &&
+			! ( isRePublicizeUpgradableViaUpsell && isRePublicizeFeatureEnabled ) ) ||
+		isPostPublished;
```

Comment:
```suggestion
		! isPostPublished;
```
Currently Publicize is disabled on draft posts. Thinking maybe this needs to be isPublicizeEnabled when post is NOT published.

Reply:
That's kind of right as we want it enabled when the post is draft, but once it's published we want to ignore the `isPublicizeEnabledMeta` value, as that relates to the main toggle, which we don't display and has no effect.

I've adjusted the logic in 96df918a1e, which appears to work in my tests, but it would be good to get some more opinions.